tools/config/__init__.py

```diff
@@ -610,24 +610,62 @@ def _get_mem_specs(self, memories, cmsis_part, exception_text):
                 continue
         raise ConfigException(exception_text)
 
-    @property
-    def rom(self):
-        """Get rom information as a pair of start_addr, size"""
+    def get_region_list(self, region='ROM'):
+        # Get list of all available regions based on memory selected in the form of list.
+        if region == 'ROM':
+            return ['IROM1', 'PROGRAM_FLASH', 'IROM2']
+        else:
+            return ['IRAM1', 'IRAM2', 'IRAM3', 'IRAM4', 'SRAM_OC', \
+                    'SRAM_ITC', 'SRAM_DTC', 'SRAM_UPPER', 'SRAM_LOWER']
+
```

Comment:
We could use get_cmsis part info and generate this table from it but that seems bit tricky (using substring to extract string) and that can get messy if new region names (which are slightly unique than usual) were added for future targets. if you can think of a better way, let me know. This seems slightly more maintainable.